csrc/dispatch.h

```diff
@@ -355,6 +355,9 @@ class TORCH_CUDA_CU_API OptOutMutator : public PolymorphicBase {
     if (mutations_.find(val) == mutations_.end()) {
       return val;
     }
+    // TODO: Check whether val is further mutated and throw error if so. This is
+    // to prevent errors where we depend on recursive mutation, which can be
+    // confusion/ambiguous to support.
```

Comment:
Not strictly necessary for this PR, but might be a good check to add?

Reply:
Yes exactly. I didn't do it yet since I wanted to verify that two-hop mutations wouldn't make sense in any of the current use cases. I believe that's true. I will add an assertion with a note that if that's needed in the future we should add an explicit interface for it then.